### DISCUSSION TITLE
Validate a version compared to a constraint

### DISCUSSION THREAD

**Discussion by steevanb** [composer/composer#9989] [Q&A / Support] [answered]
Hi,

I want to validate if a version (like `1.2.3`) compared to a constraint (like `^1.4`) could be installed.

I've not found something to do it in `composer/semver` repository, https://jubianchi.github.io/semver-check is coded in javascript and I want it in PHP.

Do you have any library who can do it?

Thanks

**Comment by stof** [comment, marked as answer]
See https://github.com/composer/semver

```php
$satisfied = \Composer\Semver\Semver::satisfies('1.2.3', '^1.4');
```

**Comment by steevanb** [comment]
Perfect, I don't know how i've missed it!

**Reply by stof** [reply]
you should accept the answer, then, to mark the question as answered.
